src/main/java/io/leonis/subra/gui/field/FieldLines.java

```diff
@@ -0,0 +1,54 @@
+package io.leonis.subra.gui.field;
+
+import com.googlecode.lanterna.*;
+import com.googlecode.lanterna.gui2.*;
+import io.leonis.subra.game.data.FieldLine;
+import java.util.Collection;
+import java.util.function.BiFunction;
+import java.util.stream.IntStream;
+import lombok.AllArgsConstructor;
+
+/**
+ * @author Jeroen de Jong
+ */
+@AllArgsConstructor
+public class FieldLines extends AbstractComponent<FieldLines> {
+
+  private final Collection<FieldLine> fieldLines;
+  private final BiFunction<Integer, Integer, TextColor> backgroundSupplier;
+
+  @Override
+  protected ComponentRenderer<FieldLines> createDefaultRenderer() {
+    return new ComponentRenderer<FieldLines>() {
+      @Override
+      public TerminalSize getPreferredSize(final FieldLines component) {
+        return null;
+      }
+
+      @Override
+      public void drawComponent(final TextGUIGraphics graphics, final FieldLines component) {
+        component.fieldLines.forEach(fieldLine ->
+            IntStream.range((int) fieldLine.getXStart(), (int) fieldLine.getXEnd())
+                .forEach(
+                    x -> IntStream.range((int) fieldLine.getYStart(), (int) fieldLine.getYEnd())
+                        .forEach(y -> {
+                          graphics.setBackgroundColor(backgroundSupplier.apply(x, y));
+                          // TODO The determination of the character should be more obvious.
+                          graphics.setCharacter(x, y, this.getCharacter(x, y, fieldLine));
+                        })));
+      }
+
+      private Character getCharacter(
```

Comment:
@RyanMeulenkamp Could you figure out a more beautiful solution for this? It's code to put either an `-`, `|` or `+` for a line.

Reply:
at least you could turn that ternary into `else if` and `else`

Reply:
It was an ternary expression; but that wasn't much clearer as well.

Reply:
no i mean

```java
        if ((fieldLine.getYStart() == y && fieldLine.getXStart() == x) ||
            (fieldLine.getYEnd() == y && fieldLine.getXEnd() == x)) {
          return '+';
        }
        else if (fieldLine.getYStart() == fieldLine.getYEnd()) {
          return '|';
        else {
          return '–';
        }
```

Reply:
I totally failed in parsing that sentence.